packages/grid/src/grid/features/useColumnMove.ts

```diff
@@ -0,0 +1,68 @@
+import { MouseEventHandler, useCallback, useRef } from "react";
+import { getAttribute } from "./getAttribute";
+import { getGridRoot } from "../model";
+import { useGridContext } from "../GridContext";
+
+export function useColumnMove() {
+  const columnMoveDataRef = useRef<{
+    unsubscribe: () => void;
+    startScreenX: number;
+    startScreenY: number;
+    startHeaderX: number;
+    startHeaderY: number;
+  }>();
+
+  const { model } = useGridContext();
+
+  const onMouseUp = useCallback(() => {
+    columnMoveDataRef.current?.unsubscribe();
+    columnMoveDataRef.current = undefined;
+    model.columnDragAndDrop.drop();
+  }, []);
+
+  const onMouseMove = useCallback((event: MouseEvent) => {
+    const { startHeaderX, startHeaderY, startScreenX, startScreenY } =
+      columnMoveDataRef.current!;
+    const shiftX = event.screenX - startScreenX;
+    const shiftY = event.screenY - startScreenY;
+    const x = startHeaderX + shiftX;
+    const y = startHeaderY + shiftY;
+    model.columnDragAndDrop.move(x, y);
+  }, []);
+
+  const onColumnMoveHandleMouseDown: MouseEventHandler<HTMLDivElement> =
+    useCallback((event) => {
```

Comment:
There is no delay or 'drag threshold' here, so column goes into drag mode even when user is just clicking column header (eg to sort)